lib/core-integration/src/Test/Integration/Scenario/API/Shelley/HWWallets.hs

```diff
@@ -332,27 +336,29 @@ spec = describe "SHELLEY_HW_WALLETS" $ do
                 (`shouldBe` restoredWalletName)
                 rl
 
-        it "The same account and mnemonic wallet can live side-by-side" $ \ctx -> do
-            mnemonics <- mnemonicToText @15 . entropyToMnemonic <$> genEntropy
+        it "The same account and mnemonic wallet can live side-by-side" $ \ctx -> runResourceT $ do
+            mnemonics <- liftIO $ mnemonicToText @15 . entropyToMnemonic <$> genEntropy
 
             -- create mnemonic wallet
             let mnemonicWalletName = "Mnemonic wallet"
             let payldCrt = payloadWith mnemonicWalletName mnemonics
-            r <- request @ApiWallet ctx (Link.postWallet @'Shelley) Default payldCrt
-            expectResponseCode @IO HTTP.status201 r
+            r1' <- postWallet ctx payldCrt
+            expectResponseCode HTTP.status201 r1'
 
             -- create from account public key
             let accXPub = pubKeyFromMnemonics mnemonics
-            _ <- restoreWalletFromPubKey @ApiWallet @'Shelley ctx accXPub restoredWalletName
+            r2' <- restoreWalletFromPubKey @ApiWallet @'Shelley ctx accXPub restoredWalletName
+
+            r1 <- request @ApiWallet ctx (Link.getWallet @'Shelley (getFromResponse id r1')) Default Empty
+            r2 <- request @ApiWallet ctx (Link.getWallet @'Shelley r2') Default Empty
 
             -- both wallets are available
-            rl <- request @[ApiWallet] ctx
-                (Link.listWallets @'Shelley) Default Empty
```

Comment:
Replaced this `listWallets` call with individually fetching the wallets once again.

IIRC, for some reason there were some wallets that hadn't already been cleaned up, that were messing up this test. And we need to adapt for when enabling `parallel` anyway.